app/models/instance.rb

```diff
@@ -15,6 +15,9 @@ def find_tenant_by_host(host)
     end
   end
 
+  VALID_HOST_NAME_REGEX = /\A[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\z/i
+  validates :host, format: { with: VALID_HOST_NAME_REGEX }, length: { maximum: 255 }
```

Comment:
@lowjoel Thanks for the detailed comments also !

So you are encouraging us to use gems as many as we can  ?

>Otherwise, I see that regex. I have to figure out what the regex means, and figure out why the regex is used. Then I add a comment saying what the regex does. Then the regex gets changed. Then the regex and the comment doesn't sync up... I leave it alone. Then the next programmer comes. Rinse and repeat.

BTW, I'm actually not too agree with this,  we can also achieve the same goal by proper name and document the function, you do not need to look at the details of regex and can just assume that the function works. This requires more application code, but makes debug easier.

> see a call saying it's validated as a hostname. I would skip it and say, okay, assume that works. If everything else fails, I check the gem (how often does this happen...?)

And here I think If we are using functions from gems, we should provide some document? 
Because to me most of time I don't know where is the function from and what it does.. normally I need to search the function name online and then hopefully I'll find the gem's document, but it's painful...